cli/stork-publisher-agent/websocket.go

```diff
@@ -149,8 +166,24 @@ func (owc *OutgoingWebsocketConnection[T]) Writer() {
 				logger.Warn().Msg("attempted to send message on closed websocket")
 				return
 			}
+
+			filteredPriceUpdates := make(SignedPriceUpdateBatch[T])
+			owc.assetIdsLock.RLock()
+			_, allAssets := owc.assetIds[WildcardSubscriptionAsset]
+			if allAssets {
+				filteredPriceUpdates = signedPriceUpdateBatch
+			} else {
+				for asset, signedPriceUpdate := range signedPriceUpdateBatch {
```

Comment:
It may be a bit of an over-optimization for now, but worth noting that since the expensive operation happening here is the signing, it is possible that some work is wasted if there are no consumers of the data for a given asset.

Reply:
ah yea that's a good point - I spent a few minutes trying to restrict the signing here but it involves some locking and information passing between objects that seems a little too complicated.

Going to keep signing everything we take in for now - in general I would assume a publisher agent will only be publishing data that someone somewhere is using, and if someone starts using a publisher agent's data for the first time I don't think the publisher would want their CPU usage to spike up suddenly. We can have predictable, sometimes inefficient processing for now and add this optimization if we find there's a need.